src/test/java/hu/elte/eserial/builder/SimpleBuilderTest.java

```diff
@@ -0,0 +1,118 @@
+    @Test
+    public void build_GivenStringTypeNullValue_ReturnsStringWithNullValue() {
+        assertNull(new SimpleBuilder(String.class).build(null));
+    }
+
+    @Test
+    public void build_GivenIntPrimitive_ReturnsIntValue() {
+        assertEquals(new Integer(1), new SimpleBuilder(int.class).build(1L));
+    }
+
+    @Test
+    public void build_GivenIntWrapper_ReturnsIntValue() {
+        assertEquals(new Integer(1), new SimpleBuilder(Integer.class).build(1L));
+    }
+
+    @Test
+    public void build_GivenShortPrimitive_ReturnsShortValue() {
+        assertEquals(new Short((short) 1), new SimpleBuilder(short.class).build(1L));
+    }
+
+    @Test
+    public void build_GivenShortWrapper_ReturnsShortValue() {
+        assertEquals(new Short((short) 1), new SimpleBuilder(Short.class).build(1L));
+    }
+
+    @Test
+    public void build_GivenBytePrimitive_ReturnsByteValue() {
+        assertEquals(new Byte((byte) 1), new SimpleBuilder(byte.class).build(1L));
+    }
+
+    @Test
+    public void build_GivenByteWrapper_ReturnsByteValue() {
+        assertEquals(new Byte((byte) 1), new SimpleBuilder(Byte.class).build(1L));
+    }
+
+    @Test
+    public void build_GivenLongPrimitive_ReturnsLongValue() {
+        assertEquals(new Long(1), new SimpleBuilder(long.class).build(1L));
+    }
+
+    @Test
+    public void build_GivenLongWrapper_ReturnsLongValue() {
+        assertEquals(new Long(1), new SimpleBuilder(Long.class).build(1L));
+    }
+
+    @Test
+    public void build_GivenFloatPrimitive_ReturnsFloatValue() {
+        assertEquals(new Float(1.5), new SimpleBuilder(float.class).build(1.5d));
+    }
+
+    @Test
+    public void build_GivenFloatWrapper_ReturnsFloatValue() {
+        assertEquals(new Float(1.5), new SimpleBuilder(Float.class).build(1.5d));
+    }
+
+    @Test
+    public void build_GivenDoublePrimitive_ReturnsDoubleValue() {
+        assertEquals(new Double(1.5), new SimpleBuilder(double.class).build(1.5d));
+    }
+
+    @Test
+    public void build_GivenDoubleWrapper_ReturnsDoubleValue() {
+        assertEquals(new Double(1.5), new SimpleBuilder(Double.class).build(1.5d));
+    }
+
+    @Test
+    public void build_GivenCharPrimitive_ReturnsChar() {
+        assertEquals(new Character('c'), new SimpleBuilder(char.class).build('c'));
+    }
+
+    @Test
+    public void build_GivenCharWrapper_ReturnsChar() {
+        assertEquals(new Character('c'), new SimpleBuilder(Character.class).build('c'));
+    }
+
+    @Test
+    public void build_GivenBooleanPrimitive_ReturnsBooleanValue() {
+        assertEquals(new Boolean("true"), new SimpleBuilder(boolean.class).build(true));
+    }
+
+    @Test
+    public void build_GivenBooleanWrapper_ReturnsBooleanValue() {
+        assertEquals(new Boolean("true"), new SimpleBuilder(Boolean.class).build(true));
+    }
+
+    @Test
+    public void build_GivenString_ReturnsStringValue() {
+        assertEquals("string", new SimpleBuilder(String.class).build("string"));
+    }
+}
```

Comment:
Most (or all?) of these could be parsed from `String`s. Test cases for those would be necessary if the parsing is implemented.